src/utils/decorators/input-validation/input-validation.js

```diff
@@ -246,6 +248,12 @@ const InputValidation = (ComposedComponent) => {
       }
     }
 
+    windowChanged = () => {
+      this.positionMessage();
+      // render to to add any eventual css rules
+      this.forceUpdate();
```

Comment:
This seems like it would be expensive. What if we changed `this.flipped` to state, and called `setState` when it changed? If not then it may be less expensive to control the flipped classes as part of the `positionMessage` function like we used to.

Reply:
Good point